src/main/java/org/spring/FullVersion/Pet.java

```diff
@@ -21,13 +20,21 @@ public Pet(Species species, String nickname) {
     }
 
     public Pet(Species species, String nickname, int age, int trickLevel, String[] habits) {
-        this.species = species;
+        this.species = Species.UNKNOWN;
```

Comment:
you should set UNKNOWN when you don't receive species, in this case your constructor receives species, but you ignore it